tests/e2e/txgen/txgen.go

```diff
@@ -195,6 +212,11 @@ func Generate(ctx context.Context, rtc client.RuntimeClient, rng *rand.Rand, acc
 						atomic.AddUint64(&okCount, 1)
 					}
 				}
+
+				// Perform an invariants check.
+				if err := CheckInvariants(ctx, rtc); err != nil {
+					errCh <- err
```

Comment:
Maybe do something like this so that if multiple goroutines report an error, this doesn't result in blocked goroutines:
```suggestion
					select {
					case errCh <- err:
					default:
					}
```

Reply:
Good catch, thanks!  I've fixed it now.